pkg/stack/service.go

```diff
@@ -2,50 +2,23 @@ package stack
 
 import (
 	"github.com/dhis2-sre/im-manager/pkg/model"
+	"golang.org/x/exp/maps"
 )
 
-type Service interface {
-	Create(stack *model.Stack) error
-	Delete(name string) error
-	CreateParameter(stack *model.Stack, parameterName string, defaultValue *string, consumed bool) (*model.StackParameter, error)
-	Find(name string) (*model.Stack, error)
-	FindAll() (*[]model.Stack, error)
-	Save(stack *model.Stack) error
-}
-
-type service struct {
-	repository Repository
-}
-
 //goland:noinspection GoExportedFuncWithUnexportedType
-func NewService(repository Repository) *service {
-	return &service{repository}
-}
-
-func (s service) Create(stack *model.Stack) error {
-	return s.repository.Create(stack)
+func NewService(stacks Stacks) *service {
+	return &service{stacks}
 }
 
-func (s service) Delete(name string) error {
-	return s.repository.Delete(name)
-}
-
-func (s service) CreateParameter(stack *model.Stack, parameterName string, defaultValue *string, consumed bool) (*model.StackParameter, error) {
-	parameter := &model.StackParameter{Name: parameterName, StackName: stack.Name, Consumed: consumed, DefaultValue: defaultValue}
-
-	err := s.repository.CreateParameter(parameter)
-
-	return parameter, err
+type service struct {
+	stacks Stacks
 }
 
 func (s service) Find(name string) (*model.Stack, error) {
-	return s.repository.Find(name)
```

Comment:
`s.repository.Find(name)` returned `errdef.NewNotFound("stack not found by name: %s", name)` if the stack is not found. I think we should still do that, or not?

Reply:
We definitely should